### PR TITLE
run dr shp-tests

### PR DESCRIPTION
**Pull request by haichangsi**
Running DR tests in a docker container for now

### CODE REVIEW

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -69,11 +69,9 @@ endif()
 add_custom_target(build-onedpl-tests
     COMMENT "Build all oneDPL tests")
 
-if (ONEDPL_USE_DR)
-    set(run-onedpl-tests-depends build-onedpl-tests shp-tests shp-tests-3)
-else()
-    set(run-onedpl-tests-depends build-onedpl-tests)
-endif()
+set(run-onedpl-tests-depends build-onedpl-tests)
```
Comment (lslusarczyk) [inline test/CMakeLists.txt:72]: why is this change?

Why builing shp-tests* was removed if ONEDPL_USE_DR is set (which means there is right C++ standard to run DR code)?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -69,11 +69,9 @@ endif()
...
+
+option(DR_DOCKER "Build shp DR tests with Docker" OFF)
```
Comment (lslusarczyk) [inline test/CMakeLists.txt:74]: What is the purpose of this option? It is not set anywhere.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -220,6 +221,24 @@ jobs:
...
+  dr-testing:
+    name: Run DR shp-tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Build docker image
+        run: docker build . -t dr-test -f .github/Dockerfile
+      - name: Run tests
+        run: |
+          docker run --name dr-test -d -v $(pwd):/repo dr-test sh -c "
```
Comment (lslusarczyk) [inline .github/workflows/ci.yml:233]: there is an option in docker "--rm" to remove container once "run" is finished, use it in CI in order to not leave rubbishes

**Review by lslusarczyk — CHANGES_REQUESTED** [review]
No test was run and even our code did not compile in latest CI run here: https://github.com/oneapi-src/oneDPL/pull/1591/checks

Last compilation error there is
```
/repo/test/distributed-ranges/shp/../../../include/oneapi/dpl/internal/distributed_ranges_impl/shp/distributed_vector.hpp:14:10: fatal error: 'oneapi/dpl/internal/distributed_ranges_impl/shp/remote_vector.hpp' file not found
   14 | #include <oneapi/dpl/internal/distributed_ranges_impl/shp/remote_vector.hpp>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
1 error generated.
gmake[3]: *** [test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/build.make:328: test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/algorithms.cpp.o] Error 1
gmake[3]: *** [test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/build.make:342: test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/containers.cpp.o] Error 1
gmake[3]: *** [test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/build.make:356: test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/copy.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:54447: test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/all] Error 2
gmake[1]: *** [CMakeFiles/Makefile2:54507: test/distributed-ranges/shp/CMakeFiles/shp-all-tests.dir/rule] Error 2
gmake: *** [Makefile:24723: shp-all-tests] Error 2
```
... but job is green.

1. Fix job to be red if compilation fails
2. fix dockers to pass 
3. manually confirm in CI logs that SHP was build and its tests passed

**Comment by haichangsi** [comment]
> No test was run and even our code did not compile in latest CI run here: https://github.com/oneapi-src/oneDPL/pull/1591/checks
> 
> Last compilation error there is
> 
> ```
> /repo/test/distributed-ranges/shp/../../../include/oneapi/dpl/internal/distributed_ranges_impl/shp/distributed_vector.hpp:14:10: fatal error: 'oneapi/dpl/internal/distributed_ranges_impl/shp/remote_vector.hpp' file not found
>    14 | #include <oneapi/dpl/internal/distributed_ranges_impl/shp/remote_vector.hpp>
>       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 1 error generated.
> 1 error generated.
> gmake[3]: *** [test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/build.make:328: test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/algorithms.cpp.o] Error 1
> gmake[3]: *** [test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/build.make:342: test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/containers.cpp.o] Error 1
> gmake[3]: *** [test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/build.make:356: test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/copy.cpp.o] Error 1
> gmake[2]: *** [CMakeFiles/Makefile2:54447: test/distributed-ranges/shp/CMakeFiles/shp-tests.dir/all] Error 2
> gmake[1]: *** [CMakeFiles/Makefile2:54507: test/distributed-ranges/shp/CMakeFiles/shp-all-tests.dir/rule] Error 2
> gmake: *** [Makefile:24723: shp-all-tests] Error 2
> ```
> 
> ... but job is green.
> 
> 1. Fix job to be red if compilation fails
> 2. fix dockers to pass
> 3. manually confirm in CI logs that SHP was build and its tests passed

fail because of a merge the previous PR with renaming a vector, I'll fix it

File: test/CMakeLists.txt (hunk 2 above)
Comment (haichangsi) [reply test/CMakeLists.txt:74]: it is used not to create new targets (build-dr-tests and run-dr-tests) to be run with others targets; for now they are run in a docker container

File: test/CMakeLists.txt (hunk 1 above)
Comment (haichangsi) [reply test/CMakeLists.txt:72]: my fault probably, I thought that C++23 is not supported and we only compile and run tests of our code in a docker container

File: test/CMakeLists.txt (hunk 2 above)
Comment (lslusarczyk) [reply test/CMakeLists.txt:74]: I still don't see sense of having this DR_DOCKER option. It is a hack. Passing responsibility to the user for determining whether DR tests will compile or not. It should depend on our checks on environment like in https://github.com/oneapi-src/oneDPL/blob/distributed-ranges/test/distributed-ranges/shp/CMakeLists.txt

Comment (haichangsi) [reply test/CMakeLists.txt]: Deleted with unused targets as we discussed

**Review by lslusarczyk — APPROVED** [review]

**Pull request merged**